### PR TITLE
fix ethics approval view

### PR DESCRIPTION
**Pull request by sanuann**
- fixes ethics approval view
- removes `schemaKey` from all array type values.

### CODE REVIEW

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetMain.vue:372]: This can be simplified to:

```suggestion
            if (each_val.hasOwnProperty('schemaKey')) {
```

I am also not sure the condition is even necessary. This first checks to see if the `schemaKey` property exists and removes it if so. I think you can skip straight to the `delete` statement and skip the conditional check.

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetMain.vue:375]:
```suggestion
```

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetMain.vue:371]: These two `_.map()` calls should be `_.forEach()` instead.

Even better would be to use the JavaScript native `Array.forEach()` function rather than lodash.

**Review by waxlamp — CHANGES_REQUESTED** [review]

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue]: what if the value is not an Object?

Comment (waxlamp) [reply src/views/DandisetLandingView/DandisetMain.vue]: It's probably not harmful to check whether it's an `Object`, but the check for `schemaKey` isn't needed.

You should keep in mind that other things in JavaScript are also objects, specifically arrays and functions.

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue]: the outer `_.map()` is on an object, not an array.

**Comment by satra** [comment]
@sanuann - since the DLP is tied to the metadata schema, you should take a look at the metadata schema to see what the different things are.

**Comment by sanuann** [comment]
> @sanuann - since the DLP is tied to the metadata schema, you should take a look at the metadata schema to see what the different things are.

is this a general comment or something to fix in my PR?

**Comment by satra** [comment]
> is this a general comment or something to fix in my PR?

a bit of a general comment so that you could check ahead of time if other elements if added to the metadata would have resulted in the same issue.

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue]: for the outer one i changed it to `_.forEach` and the inner one to `Array.forEach`

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue:372]: if I use `each_val.hasOwnProperty('schemaKey')` , I get this ESLint message:
`ESLint: Do not access Object.prototype method 'hasOwnProperty' from target object.(no-prototype-builtins)`.

How does it matter if I use `Object.keys(each_val).includes('schemaKey')` rather than `each_val.hasOwnProperty('schemaKey')` ? The first one doesn't throw a linting error too.

Comment (waxlamp) [reply src/views/DandisetLandingView/DandisetMain.vue:372]: It will be important to Google some of these topics as your work in order to build an understanding of best practices. In this case, I searched on the linting message you got and found this [documentation page](https://eslint.org/docs/rules/no-prototype-builtins) that explains the dangers of calling methods like `.hasOwnProperty()` directly, and what to do instead.

To answer your question, `Object.keys(each_val).includes()` has O(n) runtime, and though it likely will not make a difference in this case, it is better to do the standard thing: in this case you solve the performance problem, and you write code that adheres to best practices and thus will cause less burden on reviewers, other developers, and even yourself in the future if and when you come back to this code.

**Comment by waxlamp** [comment]
> > is this a general comment or something to fix in my PR?
> 
> a bit of a general comment so that you could check ahead of time if other elements if added to the metadata would have resulted in the same issue.

But you can certainly rely on the types expressed in the schema to eliminate some of the runtime typechecking in this PR too.

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue:372]: thank you for the explanation.

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetMain.vue:374]: This rule doesn't seem to apply here. Do you get an error when you remove it?

Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue:374]: yes. and i found this [documentation](https://eslint.org/docs/rules/no-param-reassign) . so perhaps it's ok to disable the rule?

Comment (waxlamp) [reply src/views/DandisetLandingView/DandisetMain.vue:374]: It looks like this [more specific version](https://eslint.org/docs/rules/no-param-reassign#props) is the thing you want here.

If you can verify that and you change it here, you can merge without another review.

**Review by waxlamp — APPROVED** [review]
Looks good, possibly with a small alteration to the eslint rule disabling statement.

File: src/views/DandisetLandingView/DandisetMain.vue
Comment (sanuann) [reply src/views/DandisetLandingView/DandisetMain.vue:374]: did you mean that I should write it this way:
```
/* eslint no-param-reassign:["error", {"ignorePropertyModificationsFor":["each_val"] }] */
delete each_val.schemaKey;
```
?

**Pull request merged**
